api/nodes/views.py

```diff
@@ -243,6 +359,62 @@ def perform_create(self, serializer):
         user = self.request.user
         serializer.save(creator=user)
 
+    # overrides ListBulkCreateUpdateDestroyAPIView
+    def bulk_update(self, request, *args, **kwargs):
+        """
+        Correctly formats bulk PUT/PATCH response
+        """
+        response = bulk_generics.ListBulkCreateUpdateDestroyAPIView.bulk_update(self, request, *args, **kwargs)
+        return Response({'data': response.data}, status=status.HTTP_200_OK)
+
+    # Overrides ListBulkCreateUpdateDestroyAPIView
+    def bulk_destroy(self, request, *args, **kwargs):
+        """
+        Handles bulk destroy of nodes. Handles permissions and enforces bulk limit.
+        """
+        user = self.request.user
+        node_list = []
+        object_type = 'nodes'
+
+        if not request.data:
+            raise ValidationError('Request must contain array of resource identifier objects.')
+
+        if not isinstance(request.data, list):
+            raise ValidationError('Expected type "list".')
+
+        for item in request.data:
+            item_type = item[u'type']
+            if item_type != object_type:
+                raise Conflict()
+            node = get_object_or_error(Node, item[u'id'], display_name='node')
+            node_list.append(node)
+            if not node.can_edit(Auth(user)):
+                raise PermissionDenied()
+        if not node_list:
+            raise NotFound()
+
+        num_items = len(node_list)
+        bulk_limit = JSONAPIListSerializer.DEFAULT_BULK_LIMIT
+
+        if num_items > bulk_limit:
+            raise ValidationError(
+                detail={'non_field_errors': ['Bulk operation limit is {}, got {}.'.format(bulk_limit, num_items)]}
+            )
+
+        self.perform_bulk_destroy(node_list)
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    # Overrides ListBulkCreateUpdateDestroyAPIView
+    def perform_destroy(self, instance):
+        user = self.request.user
+        auth = Auth(user)
+        try:
+            instance.remove_node(auth=auth)
+        except NodeStateError as err:
```

Comment:
Right, but now you're running your own permissions checks, separate from those already embedded in the relevant view function...?